### PR TITLE
Ensure that GRO reader treats lack of unit cells consistently

### LINKED ISSUE #3305: GRO reader / writer, handle missing / incomplete unitcell data
Follow up from #3132 

See comments: https://github.com/MDAnalysis/mdanalysis/pull/3132/files#r628893880, https://github.com/MDAnalysis/mdanalysis/pull/3132/files#r628897097, and https://github.com/MDAnalysis/mdanalysis/pull/3132/files#r628897541

Our default behaviour when reading / writing missing/incomplete unitcell data is both undocumented and probably inconsistent with gmx.

I would suggest here that we just be harsh, if there's missing/incomplete unitcell on reading or writing we should just throw an error and tell users what they should do to fix things (we can couple this with a transformation class that sets a unitcell that exactly accounts for the coordinates).

### PR DESCRIPTION
**Pull request by IAlibay**
Fixes #3305

Changes made in this Pull Request:
 - Add warning on reading / writing GRO files with empty box
 - Remove support for attempting to read GRO files with unit cells with neither 3 of 9 entries

PR Checklist
------------
 - [x] Tests?
 - [x] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

### CODE REVIEW

**Comment by IAlibay** [comment]
@lilyminium @mnmelo @jbarnoud I know you three have many views on this, please do review.

File: package/CHANGELOG
Comment (lilyminium) [inline package/CHANGELOG:27]:
```suggestion
  * Only GRO files with unit cells defined with 3 or 9 entries are now
    supported (Issue #3305)
```

File: package/MDAnalysis/coordinates/GRO.py
Comment (lilyminium) [inline package/MDAnalysis/coordinates/GRO.py:167]:
```suggestion
       Reader now only parses boxes defined with 3 or 9 fields.
```

**Review by lilyminium — CHANGES_REQUESTED** [review]
My brain hasn't really woken up yet, but I found those negatives hard to parse. Maybe telling the user what boxes *are* allowed is easier?

**Review by lilyminium — APPROVED** [review]

**Comment by lilyminium** [comment]
LGTM thank you @IAlibay, I'll let you decide if you want to wait on further reviews or merge

**Pull request merged**
